primary_beam_correction.py

```diff
@@ -273,11 +256,9 @@ def main():
     col = np.ravel(col)
```

Comment:
Strictly speaking, `data.shape[2]` corresponds to the x-axis of the image (the DEC axis) and `data.shape[3]` corresponds to they y-axis (the RA axis). See this: https://docs.astropy.org/en/stable/io/fits/#working-with-image-data

In your case its not a problem because your beam model is symmetric and your images are always square. But the code will likely not work as expected if these assumptions are relaxed.

Reply:
Can we discuss this more also in our meeting?